src/components/SearchBar.js

```diff
@@ -213,7 +219,9 @@ class SearchBar extends React.Component<SearchBarDefaultProps, SearchBarProps, S
 
   doSearch() {
     const searcher = this.context.searcher;
-    if (this.props.route && searcher) {
+    if (this.props.searchFunction) {
```

Comment:
This would need to be modified too, to do something similar to what happens in the `updateQuery()` method above.